include/xplot/xscales.hpp

```diff
@@ -119,6 +119,37 @@ namespace xpl
 
     using ordinal_scale = xw::xmaterialize<xordinal_scale>;
 
+    /****************************
+    * xcolor_scale declaration *
+    ****************************/
+
+    template<class D>
+    class xcolor_scale: public xscale<D>
+    {
+    public:
+
+        using base_type = xscale<D>;
+        using derived_type = D;
+
+        xcolor_scale();
+
+        xeus::xjson get_state() const;
+        void apply_patch(const xeus::xjson& patch);
+
+        XPROPERTY(std::vector<color_type>, derived_type, colors);
+        XPROPERTY(xtl::xoptional<double>, derived_type, max);
+        XPROPERTY(xtl::xoptional<double>, derived_type, mid);
+        XPROPERTY(xtl::xoptional<double>, derived_type, min);
+        XPROPERTY(X_CASELESS_STR_ENUM(linear), derived_type, scale_type, "linear");
```

Comment:
Weird that we also only have a single item in the enum :) (same in bqplot).

cc @ssunkara @dmadeka do you remember what was the reason for this?